UnitySDK/Assets/ML-Agents/Scripts/Brain.cs

```diff
@@ -94,7 +87,18 @@ private void Shutdown()
         /// </summary>
         private void BrainDecideAction()
         {
-            m_BrainBatcher?.SendBrainInfo(name, m_AgentInfos);
+            m_Communicator?.PutObservations(name, m_AgentInfos);
+            var actions = m_Communicator?.GetActions(name);
+            if (actions != null)
+                foreach (var tuple in actions)
+                {
+                    tuple.Item1.UpdateVectorAction(tuple.Item2.vectorActions);
+                    tuple.Item1.UpdateMemoriesAction(tuple.Item2.memories);
+                    tuple.Item1.UpdateTextAction(tuple.Item2.textActions);
+                    tuple.Item1.UpdateValueAction(tuple.Item2.value);
+                    tuple.Item1.UpdateCustomAction(tuple.Item2.customAction);
+                }
+
```

Comment:
Why does the brain still get to call DecideAction ? Won't it override the message of the communicator?

Reply:
Where do you think it should go?

Reply:
It's pulling the last message from the communicator every time.  So in theory, it's the same data getting updated in the agent action from the last data received from the communicator.

Reply:
I think it should either be use he communicator or use he model, not both. I don't understand your comment.

Reply:
Funny I don't understand yours either 😂